### PR TITLE
ROE-2075 Initial changes for a v1.0 model transformer

### PR DESCRIPTION
**Pull request by mwestacott**
### JIRA link

https://companieshouse.atlassian.net/browse/ROE-2075

### Change description

* New transformer class added to handle version 1.0 models
* Unit and integration tests added

### Work checklist

- [ ] Tests added where applicable

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/converter/Version1_0Transformer.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package uk.gov.companieshouse.overseasentitiesapi.converter;
+
+import org.bson.Document;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.overseasentitiesapi.model.SchemaVersion;
+import uk.gov.companieshouse.overseasentitiesapi.utils.ApiLogger;
+
```
Comment (markpit) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/converter/Version1_0Transformer.java:7]: i'd add some class level comments to explain what is changed/moved during a version 1.0 transform

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/converter/Version1_0Transformer.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+@Component
+public class Version1_0Transformer implements DocumentTransformer {
```
Comment (markpit) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/converter/Version1_0Transformer.java:9]: I'd rename class to Version_1_0_DocumentTransformer

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/converter/Version1_0Transformer.java (hunk 1 above)
Comment (mwestacott) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/converter/Version1_0Transformer.java:7]: Yep. Good suggestion.

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/converter/Version1_0Transformer.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    }
+
+    @Override
+    public void transform(Document submissionDocument) {
+        ApiLogger.info("Transforming a version 1.0 document...");
+
+        Document entityDocument = (Document) submissionDocument.get(ENTITY_BLOCK_NAME);
+        String entityName = entityDocument.getString(OLD_ENTITY_NAME_FIELD_NAME);
+        Document entityNameDocument = new Document();
+        entityNameDocument.put(NEW_ENTITY_NAME_BLOCK_NAME, entityName);
```
[the remark this answers is not in the document]
Comment (mwestacott) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/converter/Version1_0Transformer.java:28]: Good spot. Yes, need reversing but then we don't need a constant for the new entity field name (as same as before). Will push a change.

**Review by markpit — APPROVED** [review]

**Review by mattch1 — APPROVED** [review]

**Pull request merged**